src/seedu/addressbook/data/person/Exam.java

```diff
@@ -77,41 +81,42 @@ public Exam(String examName, String subjectName, String examDate, String examSta
-        this.examEndTime = toEdit.getExamEndTime();
-        this.examDetails = toEdit.getExamDetails();
-        this.isPrivate = toEdit.isPrivate();
-
-        for (Map.Entry<String, String> entry : changedDetails.entrySet()) {
-            String attribute = entry.getKey();
+    public Exam(Exam toEdit, Map<ExamField, String> changedDetails) throws IllegalValueException {
+        examName = toEdit.getExamName();
+        subjectName = toEdit.getSubjectName();
+        examDate = toEdit.getExamDate();
+        examStartTime = toEdit.getExamStartTime();
+        examEndTime = toEdit.getExamEndTime();
+        examDetails = toEdit.getExamDetails();
+        takers = toEdit.getTakers();
+        isPrivate = toEdit.isPrivate();
+
+        for (Map.Entry<ExamField, String> entry : changedDetails.entrySet()) {
+            ExamField attribute = entry.getKey();
             String newValue = entry.getValue();
-            if ("examName".equals(attribute)) {
-                this.examName = newValue.trim();
-            } else if ("subjectName".equals(attribute)) {
-                this.subjectName = newValue.trim();
-            } else if ("examDate".equals(attribute)) {
+            if (ExamField.examName.equals(attribute)) {
+                examName = newValue.trim();
+            } else if (ExamField.subjectName.equals(attribute)) {
+                subjectName = newValue.trim();
+            } else if (ExamField.examDate.equals(attribute)) {
                 if (!isValidDate(newValue.trim())) {
                     throw new IllegalValueException(MESSAGE_DATE_CONSTRAINTS);
                 }
-                this.examDate = newValue.trim();
-            } else if ("examStartTime".equals(attribute)) {
+                examDate = newValue.trim();
+            } else if (ExamField.examStartTime.equals(attribute)) {
                 if (!isValidTime(newValue.trim())) {
                     throw new IllegalValueException(MESSAGE_TIME_CONSTRAINTS);
                 }
-                this.examStartTime = newValue.trim();
-            } else if ("examEndTime".equals(attribute)) {
+                examStartTime = newValue.trim();
+            } else if (ExamField.examEndTime.equals(attribute)) {
                 if (!isValidTime(newValue.trim())) {
                     throw new IllegalValueException(MESSAGE_TIME_CONSTRAINTS);
                 }
-                this.examEndTime = newValue.trim();
-            } else if ("examDetails".equals(attribute)) {
-                this.examDetails = newValue.trim();
-            } else if ("isPrivate".equals(attribute)) {
-                this.isPrivate = "y".equals(newValue.trim());
+                examEndTime = newValue.trim();
+            } else if (ExamField.examDetails.equals(attribute)) {
+                examDetails = newValue.trim();
+            } else if (ExamField.isPrivate.equals(attribute)) {
+                isPrivate = "y".equals(newValue.trim());
             }
```

Comment:
Not sure if a case switch will be cleaner here

Reply:
i dont have a default. unless throw an exception?

Reply:
Hmm is there a coding standard that requires all switch statements to have a default case?

Reply:
https://oss-generic.github.io/process/codingStandards/CodingStandard-Java.html#layout

Reply:
Hmm I'm not sure if we are to interpret that as enforcing the inclusion of default case. If it really is required, I guess asserting false would make more sense than a Exception